sql/src/main/java/io/crate/analyze/WhereClause.java

```diff
@@ -89,19 +92,23 @@ public WhereClause normalize(EvaluatingNormalizer normalizer, TransactionContext
         if (nullReplacedQuery == query) {
             return this;
         }
-        return new WhereClause(nullReplacedQuery, docKeys.orElse(null), partitions, clusteredBy.orElse(null));
+        return new WhereClause(nullReplacedQuery, docKeys.orElse(null), partitions, clusteredBy);
+    }
+
+    public boolean usesClusterByColumns() {
```

Comment:
Given that we expose the whole underlying `Set<Symbol>` as well I'd avoid having a separate property for this as it creates a contract that is not visible (`usesClusteredByColumns` implies that `clusteredBy` is non-empty. This is a bit of an implementation detail leak.